gwcs/wcs.py

```diff
@@ -390,10 +393,44 @@ def pipeline(self):
         return self._pipeline
 
     @property
+    def bounding_box(self):
+        """
+        Return the range of acceptable values for each input axis.
+        The order of the axes is `~gwcs.coordinate_frames.CoordinateFrame.axes_order`.
+        """
+        frames = self.available_frames
+        transform_0 = self.get_transform(frames[0], frames[1])
+        try:
+            # Model.bounding_box is in numpy order, need to reverse it first.
+            bb = transform_0.bounding_box[::-1]
+        except NotImplementedError:
+            return None
+        bb = np.array(bb)[np.array(self.input_frame.axes_order)]
+        return tuple(tuple(item) for item in bb)
+
+    @bounding_box.setter
+    def bounding_box(self, value):
+        """
+        Set the range of acceptable values for each input axis.                                                              
+        The order of the axes is `~gwcs.coordinate_frames.CoordinateFrame.axes_order`.
+
+        Parameters
+        ----------
+        value : tuple
+            Tuple of tuples with ("low", high") values for the range.
+        """
+        frames = self.available_frames
+        transform_0 = self.get_transform(frames[0], frames[1])
+        axes_ind = np.argsort(self.input_frame.axes_order)
+        transform_0.bounding_box = np.array(value)[axes_ind][::-1]
+        self.set_transform(frames[0], frames[1], transform_0)
+
+    @property
     def domain(self):
         """
         Return the range of acceptable values for each input axis.
         """
+        warnings.warn('"domain" was deprecated in v0.8 and will be removed in the next version. Use "bounding_box" instead.')
```

Comment:
You may want to use `DepreciationWarning` category here and below on line 446.  I.e.:
```
warnings.warn(message, DeprecationWarning)
```
Just in case anyone wants to filter these warnings for some reason.